health_checks.js

```diff
@@ -0,0 +1,81 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2018, Alert Logic, Inc
+ * @doc
+ *
+ * Halth check functions common for all collectors
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+
+const AWS = require('aws-sdk');
+const async = require('async');
+
+function checkCloudFormationStatus(event, context, callback) {
+    var stackName = event.StackName;
+    var cloudformation = new AWS.CloudFormation();
+    cloudformation.describeStacks({StackName: stackName}, function(err, data) {
+        if (err) {
+            return callback(errorMsg('AWS00001', stringify(err)));
+        } else {
+            var stackStatus = data.Stacks[0].StackStatus;
+            if (stackStatus === 'CREATE_COMPLETE' ||
+                stackStatus === 'UPDATE_COMPLETE' ||
+                stackStatus === 'UPDATE_IN_PROGRESS' ||
+                stackStatus === 'UPDATE_ROLLBACK_COMPLETE' ||
+                stackStatus === 'UPDATE_ROLLBACK_IN_PROGRESS' ||
+                stackStatus === 'UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS' ||
+                stackStatus === 'REVIEW_IN_PROGRESS') {
+                return callback(null);
+            } else {
+                return callback(errorMsg('AWS00002', 'CF stack has wrong status: ' + stackStatus));
+            }
+        }
+    });
+}
+
+function get_health_check(event, context, customChecks, callback) {
+    customChecksFuns = customChecks.map(f => function(asyncCallback) {
+        f(event, context, asyncCallback)
+    });
+    async.waterfall([
```

Comment:
It will be faster to run all healthcheck in parallel. Please use `async.map` here.